sql/lanterndb.sql

```diff
@@ -9,25 +9,48 @@ COMMENT ON ACCESS METHOD hnsw IS 'LanternDB vector index access method. Can be c
 
 
 -- functions
-
+CREATE FUNCTION ldb_generic_dist(real[], real[]) RETURNS real
+	AS 'MODULE_PATHNAME' LANGUAGE C IMMUTABLE STRICT PARALLEL SAFE;
+	
+CREATE FUNCTION ldb_generic_dist(integer[], integer[]) RETURNS real
+	AS 'MODULE_PATHNAME' LANGUAGE C IMMUTABLE STRICT PARALLEL SAFE;
+	
 CREATE FUNCTION l2sq_dist(real[], real[]) RETURNS real
 	AS 'MODULE_PATHNAME' LANGUAGE C IMMUTABLE STRICT PARALLEL SAFE;
 
+CREATE FUNCTION cos_dist(real[], real[]) RETURNS real
+	AS 'MODULE_PATHNAME' LANGUAGE C IMMUTABLE STRICT PARALLEL SAFE;
 
--- operators
+CREATE FUNCTION hamming_dist(integer[], integer[]) RETURNS integer
+	AS 'MODULE_PATHNAME' LANGUAGE C IMMUTABLE STRICT PARALLEL SAFE;
 
+-- operators
 CREATE OPERATOR <-> (
-	LEFTARG = real[], RIGHTARG = real[], PROCEDURE = l2sq_dist,
+	LEFTARG = real[], RIGHTARG = real[], PROCEDURE = ldb_generic_dist,
 	COMMUTATOR = '<->'
 );
 
+CREATE OPERATOR <-> (
+	LEFTARG = integer[], RIGHTARG = integer[], PROCEDURE = ldb_generic_dist,
+	COMMUTATOR = '<->'
+);
 
 -- operator classes
 CREATE OPERATOR CLASS ann_l2_ops
```

Comment:
what does `ann` stand for in these operator classes? This is actually something that the clients will type out when creating indexes so we should be thoughtful when naming.
maybe `vec_l2sq_ops`,  `dist_l2sq_ops`?

Reply:
it stands for approximate nearest neighbors, but yes I also like `vec_l2sq_ops` one